src/app/forms/validators/custom-async-validators.ts

```diff
@@ -114,7 +117,52 @@ export class CustomAsyncValidators {
 
           return null;
         })
+   
+        );
+  }
+
+  static hideIfEqualsWithCondition(store: Store<State>, sibling: string, value: string, conditions: Condition | Condition[]): AsyncValidatorFn {
+    return (control: AbstractControl): Observable<ValidationErrors | null> =>
+      store.pipe(
+        take(1),
+        select(testResultInEdit),
+        map(testResult => {
+          if (!testResult || !control?.parent) { return null; }
+
+          const siblingControl = control.parent.get(sibling) as CustomFormControl;
+
+          const conditionsPassed = CustomAsyncValidators.checkConditions(testResult, conditions);
+
+          siblingControl.meta.hide = conditionsPassed && (Array.isArray(value) ? value.includes(control.value) : control.value === value);
+
+          return null;
+        })
       );
   }
- 
+
+  static checkConditions(testResult: TestResultModel, conditions: Condition | Condition[] ){
+    if (!Array.isArray(conditions)) {
+      return CustomAsyncValidators.checkCondition(testResult, conditions);
+    }
+
+    for (const condition of conditions) {
+      if(!CustomAsyncValidators.checkCondition(testResult, condition)) {
+        return false;
+      }
+    }
+    
+    return true;
+  }
+
+  static checkCondition(testResult: TestResultModel, condition: Condition ){
+    const {field, operator, value} = condition
+
+    const fieldValue = testResult.testTypes[0].hasOwnProperty(field)
+      ? (testResult?.testTypes[0] as any)[field]
```

Comment:
can this be null after you've already checked the condition on line 160 ?

Reply:
Hi Tom good point I will delete the question mark after testResult as I guess it can't be null if line 160 succeeds. Thanks

Reply:
This change has been made.